ui/apps/platform/src/Components/CompoundSearchFilter/components/ConditionNumber.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { useEffect, useState } from 'react';
+import { Button, NumberInput, SelectOption } from '@patternfly/react-core';
+import { ArrowRightIcon } from '@patternfly/react-icons';
+
+import { ensureString } from '../utils/utils';
+
+import SimpleSelect from './SimpleSelect';
+
+export type ConditionNumberProps = {
+    value: string;
+    onSearch: (value: string) => void;
+};
+
+const conditionMap = {
+    'Is greater than': '>',
+    'Is greater than or equal to': '>=',
+    'Is equal to': '=',
+    'Is less than or equal to': '<=',
+    'Is less than': '<',
+};
+
+const conditions = Object.keys(conditionMap);
+
+const minValue = 0;
+const maxValue = 10;
+const incrementValue = 0.1;
+
+function roundToOneDecimalPlace(value: number) {
+    return parseFloat(value.toFixed(1));
+}
+
+const normalizeBetween = (value: number, min: number, max: number) => {
+    if (min !== undefined && max !== undefined) {
+        return Math.max(Math.min(value, max), min);
+    }
+    if (value <= min) {
+        return min;
+    }
+    if (value >= max) {
+        return max;
+    }
+    return value;
+};
+
+function getDefaultCondition() {
+    return conditions[0];
+}
+
+function getDefaultNumber() {
+    return minValue;
+}
+
+function valueToConditionNumber(value: string) {
+    const [newConditionValue, newNumber] = value.split(' ');
+    const newCondition =
+        conditions.find((condition) => {
+            return conditionMap[condition] === newConditionValue;
+        }) || conditions[0];
+    return {
+        newCondition,
+        newNumber: Number(newNumber) || minValue,
+    };
+}
+
+function conditionNumberToValue(conditionLabel: string, number: number): string {
+    const conditionValue = conditionMap[conditionLabel];
+    const newValue = `${conditionValue} ${number}`;
+    return newValue;
+}
+
+function ConditionNumber({ value, onSearch }: ConditionNumberProps) {
+    const [selectedCondition, setSelectedCondition] = useState(() => getDefaultCondition());
+    const [number, setNumber] = useState(() => getDefaultNumber());
+
+    useEffect(() => {
+        const { newCondition, newNumber } = valueToConditionNumber(value);
+        setSelectedCondition(newCondition);
+        setNumber(newNumber);
+    }, [value]);
```

Comment:
My initial thought here was how we could remove this `useEffect`. In general, if an effect is triggered by some user action I think it is better to have whatever side effect would occur happen in the event callback or child function call rather than in the next render cycle. The only exception I've found to this so far is when we want to trigger some code when a user navigates to a page, as we don't have an easy way to hook into the navigation event.

On second glance, I'm not sure what the purpose of the code in the effect is for. Is the goal to parse the input `value` into a correct condition/number? If so, is that something we could do directly in render instead? Although it does seem like the component works without the effect altogether.

Reply:
I was intending to keep track of one state to make things easier, but looking into it more, I think this definitely could be improved. Let me consider your suggestion and see if we can do things a bit differently here

Reply:
Ok, I made a few changes. I'm now tracking `value` as `{ condition: string; number: number }` when the `inputType` is `condition-number`. This should simplify things. We should review the logic for the `onSearch` callback in `CompoundSearchFilterInputField` in light of these changes. Essentially, we'll need to examine the `value` type and determine how to translate it into the format we want to maintain in the URL search.